app/controllers/ResubmitAReturnController.scala

```diff
@@ -73,11 +75,14 @@ class ResubmitAReturnController @Inject() (
                 BadRequest(view(formWithErrors, periods))
             }
           },
-        value =>
+        value => {
+          val periodKey = PeriodKey(value.key)
           for {
-            updatedAnswers <- Future.fromTry(request.userAnswers.set(ResubmitAReturnPage, value))
-            _              <- sessionRepository.set(updatedAnswers)
-          } yield Redirect(navigator.nextPage(ResubmitAReturnPage, NormalMode, updatedAnswers))
+            updatedAnswers <- Future.fromTry(request.userAnswers.set(ResubmitAReturnPage(periodKey), value))
+            userAnswersOpt <- previousReturnsService.convertReturnToUserAnswers(PeriodKey(value.key), updatedAnswers)
```

Comment:
@wp94 Can we pass the val period ke
```suggestion
            userAnswersOpt <- previousReturnsService.convertReturnToUserAnswers(periodKey, updatedAnswers)
```